mm2src/coins/coin_errors.rs

```diff
@@ -0,0 +1,442 @@
+    fn from(err: JsonRpcError) -> Self { Self::JsonRpcError(err) }
+}
+
+impl From<UtxoRpcError> for CheckPaymentSentError {
+    fn from(err: UtxoRpcError) -> Self { Self::UtxoRpcError(err) }
+}
+
+impl From<AddressParseError> for CheckPaymentSentError {
+    fn from(err: AddressParseError) -> Self { Self::AddressParseError(err.to_string()) }
+}
+
+impl From<serialization::Error> for CheckPaymentSentError {
+    fn from(err: serialization::Error) -> Self { Self::DeserializationErr(err.to_string()) }
+}
+
+impl From<UnexpectedDerivationMethod> for CheckPaymentSentError {
+    fn from(err: UnexpectedDerivationMethod) -> Self { Self::UnexpectedDerivationMethod(err) }
+}
+
+#[derive(Debug, Display, PartialEq)]
+pub enum ExtractSecretError {
+    DeserializationErr(String),
+    DecodingError(String),
+    ExtractionFailed(String),
+    CouldNotObtainSecret(String),
+    #[display(fmt = "Invalid arguments in 'receiverSpend' call: {:?}", _0)]
+    InvalidArguments(Vec<ethabi::Token>),
+    #[display(fmt = "Expected secret to be fixed bytes, decoded function data is {:?}", _0)]
+    ExpectedFixedBytes(Vec<ethabi::Token>),
+}
+
+impl From<ethabi::Error> for ExtractSecretError {
+    fn from(err: ethabi::Error) -> Self { Self::DecodingError(err.to_string()) }
+}
+
+impl From<rlp::DecoderError> for ExtractSecretError {
+    fn from(err: rlp::DecoderError) -> Self { Self::DecodingError(err.to_string()) }
+}
+
+impl From<serialization::Error> for ExtractSecretError {
+    fn from(err: serialization::Error) -> Self { Self::DeserializationErr(err.to_string()) }
+}
+
+#[derive(Debug, Display, PartialEq)]
+pub enum MyAddressError {
+    AddressParseError(AddressParseError),
+    AddrDisplayError(String),
+    #[display(fmt = "DeprecatedWalletAddr: 'my_address' is deprecated for HD wallets")]
+    DeprecatedWalletAddr,
+    EncodingError(String),
+    #[display(fmt = "Invalid address: {}", _0)]
+    InvalidAddress(String),
+    Internal(String),
+    UnsupportedAddr(String),
+    MethodNotSupported(String),
+    ScriptHashTypeNotSupported {
+        script_hash_type: String,
+    },
+    CashAddressErr(String),
+    #[display(fmt = "Transaction Reading Error: {}", _0)]
+    TxReadError(String),
+    UnexpectedDerivationMethod(UnexpectedDerivationMethod),
+    UtxoRpcError(String),
+}
+
+impl From<AddressParseError> for MyAddressError {
+    fn from(e: AddressParseError) -> Self { MyAddressError::AddressParseError(e) }
+}
+
+impl From<UnexpectedDerivationMethod> for MyAddressError {
+    fn from(e: UnexpectedDerivationMethod) -> Self { MyAddressError::UnexpectedDerivationMethod(e) }
+}
+
+impl From<keys::Error> for MyAddressError {
+    fn from(e: keys::Error) -> Self { MyAddressError::Internal(e.to_string()) }
+}
+
+impl From<UnsupportedAddr> for MyAddressError {
+    fn from(e: UnsupportedAddr) -> Self { MyAddressError::UnsupportedAddr(e.to_string()) }
+}
+
+#[derive(Debug, Display, PartialEq)]
+pub enum AddressParseError {
```

Comment:
The same problem with `AddressParseError`. It definitely contains multiple unnecessary error variants. I can't say that there should be only two variants, but there shouldn't be so many ones.